modin/pandas/dataframe.py

```diff
@@ -819,12 +819,57 @@ def explode(
             pandas.DataFrame.explode, column, ignore_index=ignore_index
         )
 
+    def _update_var_dicts_in_kwargs(self, expr, **kwargs):
+        """
+        Copy variables with "@" prefix in `local_dict` and `global_dict`.
+
+        Parameters
+        ----------
+        expr : str
+            The expression string to search variables with "@" prefix.
+        **kwargs : dict
+            See the documentation for eval() for complete details on the keyword arguments accepted by query().
+
+        Returns
+        -------
+        kwargs : dict
+            Dict with updated `local_dict` and `global_dict` keys.
+        """
+        import re
```

Comment:
Imports should be done in the top of the file unless there's very good reason not to.
They implicitly acquire import lock which could cause a deadlock in some very funny code, plus it might confuse some IDEs for completion etc.